code/framework/encryption/src/main/java/io/cattle/platform/framework/encryption/EncryptionUtils.java

```diff
@@ -0,0 +1,52 @@
+package io.cattle.platform.framework.encryption;
+
+import java.io.FileReader;
+import java.io.FileWriter;
+import java.io.IOException;
+
+import org.json.simple.JSONObject;
+import org.json.simple.parser.JSONParser;
+import org.json.simple.parser.ParseException;
+
+public class EncryptionUtils {
+    private static String fileName = "keys.json";
+
+    public static boolean isEqual(byte[] a, byte[] b) {
+        if (a.length != b.length) {
```

Comment:
On a second thought, if the filed-to-encrypt/decrypt is null, we shouldn't even try to call encryption methods on it. Basically upon the very first retrieval of the field, check if its not null, then if it is, return. If not null, check if encryption is enabled for it, and proceed with the logic.